alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/SummaryMessageContentProcessor.java

```diff
@@ -105,60 +109,78 @@ private ProviderMessageContent summarize(final ProviderMessageContent message) {
         return itemsByOperation;
     }
 
-    private LinkedHashSet<ComponentItem> createSummarizedComponentItems(final Set<ComponentItem> componentItemsForOperation) {
-        final LinkedHashSet<ComponentItem> summarizedCategoryItems = new LinkedHashSet<>();
-        final Map<String, ComponentAttributeMap> itemsByShallowKey = collectComponentItemData(componentItemsForOperation);
-        for (final ComponentItem componentItem : componentItemsForOperation) {
-            String summaryKey = createSummaryKey(componentItem.getComponentKeys(), componentItem.getItemsOfSameName());
-            final SortedSet<LinkableItem> summarizedLinkableItems = createSummarizedLinkableItems(itemsByShallowKey.get(summaryKey));
-            try {
-                if (!summarizedLinkableItems.isEmpty()) {
-                    ComponentItem newComponentItem = messageOperationCombiner.createNewComponentItem(componentItem, summarizedLinkableItems);
-                    summarizedCategoryItems.add(newComponentItem);
-                }
-            } catch (AlertException e) {
-                // If this happens, it means there is a bug in the Collector logic.
-                throw new AlertRuntimeException(e);
-            }
+    private LinkedHashSet<ComponentItem> summarizeCategoryComponentCounts(Set<ComponentItem> componentItems) {
+        Map<String, Map<ComponentItemPriority, Set<ComponentItem>>> itemsByCategory = new LinkedHashMap<>();
+
+        for (final ComponentItem item : componentItems) {
+            Map<ComponentItemPriority, Set<ComponentItem>> itemsByPriority = itemsByCategory.computeIfAbsent(item.getCategory(), ignored -> new LinkedHashMap<>());
+            Set<ComponentItem> itemList = itemsByPriority.computeIfAbsent(item.getPriority(), ignored -> new LinkedHashSet<>());
+            itemList.add(item);
         }
 
-        return summarizedCategoryItems;
-    }
+        LinkedHashSet<ComponentItem> componentSummaries = new LinkedHashSet<>();
+        for (Map.Entry<String, Map<ComponentItemPriority, Set<ComponentItem>>> entry : itemsByCategory.entrySet()) {
+            for (Map.Entry<ComponentItemPriority, Set<ComponentItem>> priorityEntries : entry.getValue().entrySet()) {
+                try {
+                    Set<ComponentItem> items = priorityEntries.getValue();
 
-    private Map<String, ComponentAttributeMap> collectComponentItemData(Set<ComponentItem> componentItemsForOperation) {
-        Map<String, ComponentAttributeMap> itemsByCategory = new LinkedHashMap<>();
-        for (final ComponentItem categoryItem : componentItemsForOperation) {
-            ComponentAttributeMap itemsOfSameName = categoryItem.getItemsOfSameName();
-            String summaryKey = createSummaryKey(categoryItem.getComponentKeys(), itemsOfSameName);
+                    List<LinkableItem> attributeSet = items.stream()
```

Comment:
This variable is not a set, so we should call it something like `summarizableAttributes`.